### PR TITLE
Add vtkbool recipe

### PR DESCRIPTION
**Pull request by whophil**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by whophil** [comment]
@akaszynski do you have any interest in being a maintainer?

**Comment by akaszynski** [comment]
@whophil, yep. Sign me up.

**Comment by whophil** [comment]
@conda-forge/help-c-cpp, ready for review!

Question - the source repo only has one release, which is listed as a RC. Is this acceptable?

**Comment by wolfv** [comment]
LGTM. Regarding the release -- I checked the repo and the author seems to be active. Maybe you can ask for a 3.0 or 3.1 release? Or do you want to merge like this before? I am not sure the `rc` ordering will fine, I'd have to check the conda spec again.

**Review by carterbox — CHANGES_REQUESTED** [review]
You need to wait until a non-rc-release or put this package into an rc channel via conda_build_config.yaml 

https://conda-forge.org/docs/maintainer/knowledge_base.html#pre-release-builds

Please ping the team again after either of those things has happened.

File: recipes/vtkbool/meta.yaml
Comment (carterbox) [inline recipes/vtkbool/meta.yaml:15]:
```suggestion
build:
  number: 0
  run_exports:
    - {{ pin_subpackage( vtkbool, max_pin='x.x.x' ) }}
```
Because you're shipping headers, you need a run_export which ensures that packages which build against this package get a compatible version at runtime. Set the max_pin according to ABI guarantees. Assume 'x.x.x' if there are no guarantees.

File: recipes/vtkbool/meta.yaml
Comment (carterbox) [inline recipes/vtkbool/meta.yaml:33]:
```suggestion
test:
  commmands:
    - test -f $PREFIX/include/vtkBoolModule.h  # [unix]
    - test -f $PREFIX/lib/vtkBool${SHLIB_EXT}  # [unix]
  imports:
    - vtkbool
```
https://conda-forge.org/docs/maintainer/adding_pkgs.html#simple-existence-tests

Add tests which indicate which types of artifacts are packaged.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by whophil** [comment]
@carterbox @wolfv this is ready for re-review, whenever you have a chance. Thanks!

**Comment by carterbox** [comment]
@conda-forge/help-c-cpp please review

**Comment by carterbox** [comment]
Ooops. I just wanted to update this branch. 🤦🏼

**Comment by carterbox** [comment]
Recipe looks good!

**Pull request merged**
